src/handlers/private-notifications/comment-triggers.ts

```diff
@@ -0,0 +1,184 @@
+import { Bot } from "../../bot";
+import { Context, SupportedEvents } from "../../types";
+import { StorageUser } from "../../types/storage";
+import { CallbackResult } from "../../types/proxy";
+import { TelegramBotSingleton } from "../../types/telegram-bot-single";
+import { logger } from "../../utils/logger";
+import { PluginContext } from "../../types/plugin-context-single";
+
+const reminderCommentRegex = /@(\w+), this task has been idle for a while. Please provide an update./gi;
+const base64ClaimUrlRegex = /href="https:\/\/[^/]+\/?\?claim=([A-Za-z0-9+/=]+)"/gi;
+const amountPatternRegex = /\[\s*\d+(\.\d+)?\s*[A-Z]+\s*\]/gi;
+const githubUsernameRegex = /<h6>@([a-zA-Z0-9-]{1,39})<\/h6>/gi;
+
+// we'll have multiple permit comments to parse out here
+// the regex is capturing the claim url and the github username
+
+export async function notificationsRequiringComments(
+  context: Context<"issue_comment.created" | "issue_comment.edited", SupportedEvents["issue_comment.created" | "issue_comment.edited"]>
+): Promise<CallbackResult> {
+  const {
+    adapters: { storage },
+    payload,
+    logger,
+  } = context;
+
+  const ctxInstance = PluginContext.getInstance();
+
+  const body = payload.comment.body;
+  const paymentComment = body.match(base64ClaimUrlRegex) && body.match(amountPatternRegex) && body.match(githubUsernameRegex);
```

Comment:
This obviously is not a great approach but until we have `structuredMetadata` and a `structuredMetadataParser` it's going to be unclean. I'll need my reviewers to be test subjects and register for payment notifications and I can optimize if need be.

I could pull just the base64 strings, decode them and match the wallet against the DB. Problem is, good chance of false flags.

We can iterate on the notifications logic but we need to test in prod with this plugin it has shown.